controllers/k8sgpt_controller.go

```diff
@@ -124,7 +124,7 @@ func (r *K8sGPTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	// Check and see if the instance is new or has a K8sGPT deployment in flight
 	deployment := v1.Deployment{}
 	err = r.Get(ctx, client.ObjectKey{Namespace: k8sgptConfig.Namespace,
-		Name: "k8sgpt-deployment"}, &deployment)
```

Comment:
This is a sensible move, but yes we need to make people aware of this

Reply:
We could inform the adopters by highlighting this change in the `CHANGELOG.md` file, as well as in the `README.md` by bumping up the minor version of the operator version: a "migration" guide could be shared in the changelog file, getting more visibility about the change to whoever's updating.

The commit has been already marked as _breaking_, although it's not entirely breaking, but rather, it changes the naming convention for the deployed resources, thus creating some orphaned resources.

Do you think this could be enough?

Reply:
Shouldn't we handle all these changes by updating the version of the k8sgpt resource?


```yaml
apiVersion: core.k8sgpt.ai/v1alpha2
kind: K8sGPT
```

This would allow us to maintain two versions of the controller, preventing any breaking changes and avoiding complicating the migration process. However, it will introduce "dead" code within the controller itself.

Reply:
This is a very good point